### PR TITLE
[ILM] Add Freeze Action

### LINKED ISSUE #34630: [ILM] Add ILM action for freezing an index
When frozen indices are implemented (https://github.com/elastic/elasticsearch/issues/34352) we should add a new action to ILM so a user can freeze indices in the cold phase. The actual implementation of the action will depend on the final API for frozen indices.

### PR DESCRIPTION
**Pull request by talevy**
This commit adds a new ILM Action for
freezing indices in the cold phase.

Closes #34630.

### CODE REVIEW

**Comment by talevy** [comment]
run the gradle build tests 1

File: docs/reference/ilm/policy-definitions.asciidoc [hunk 1]
```diff
@@ -284,6 +284,31 @@ PUT _ilm/policy/my_policy
 --------------------------------------------------
 // CONSOLE
 
+[[ilm-freeze-action]]
+==== Freeze
+
+Phases allowed: cold.
+
+This action will <<frozen-indices, freeze>> the index
+by calling the <<freeze-index-api, Freeze Index API>>.
```
Comment (gwbrown) [inline docs/reference/ilm/policy-definitions.asciidoc]: This should have a note similar to the [freeze index API docs](https://www.elastic.co/guide/en/elasticsearch/reference/6.x/freeze-index-api.html) that calls out that this action closes and opens the index, and that has repercussions (index will not be queryable), as that's different from the other actions.

File: x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeSeriesLifecycleActionsIT.java [hunk 2]
```diff
@@ -423,6 +426,21 @@ public void testShrinkAction() throws Exception {
...
+        createNewSingletonPolicy("cold", new FreezeAction());
+        updatePolicy(index, policy);
+        assertBusy(() -> {
+            Map<String, Object> settings = getOnlyIndexSettings(index);
+            assertThat(getStepKeyForIndex(index), equalTo(TerminalPolicyStep.KEY));
+            assertThat(settings.get(IndexMetaData.INDEX_BLOCKS_WRITE_SETTING.getKey()), equalTo("true"));
+            assertThat(settings.get(IndexSettings.INDEX_SEARCH_THROTTLED.getKey()), equalTo("true"));
+            assertThat(settings.get(FrozenEngine.INDEX_FROZEN.getKey()), equalTo("true"));
+        });
+
```
Comment (gwbrown) [inline x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeSeriesLifecycleActionsIT.java]: Extra newline

**Review by AthenaEryma — APPROVED** [review]
Left a couple nits, otherwise LGTM with the understanding that https://github.com/elastic/elasticsearch/issues/37085 should be addressed before the next release as well.

**Comment by talevy** [comment]
thanks, @gwbrown, for the recommendations. I've copied over the important message around freezing into the action docs

**Pull request merged**